src/swap/OddEven.jl

```diff
@@ -0,0 +1,49 @@
+abstract type AbstractOddEven end
```

Comment:
Seems weird to create an abstract type with a single struct descendant to it, then ending up using `Union{AbstractOddEven, VariationalOddEven}` later.. I am generally wary of creating types that are not used in the code.